osidb/tests/test_search.py

```diff
@@ -309,3 +309,70 @@ def test_search_flaws_by_similar_cve(self, auth_client, test_api_uri):
         body = response.json()
         assert body["count"] == 1
         assert body["results"][0]["cve_id"] == "CVE-2001-0414"
+
+    def test_search_flaws_by_query(self, auth_client, test_api_uri):
+        """Test searching flaws by djangoql query."""
+        response = auth_client().get(f"{test_api_uri}/flaws")
+        assert response.status_code == 200
+        body = response.json()
+        assert body["count"] == 0
+
+        FlawFactory(
+            title="title",
+            comment_zero="comment_zero",
+            cve_description="cve_description",
+            embargoed=False,
+            statement="statement",
+            cve_id="CVE-2001-0414",
+        )
+        FlawFactory(
+            title="other flaw",
+            comment_zero="comment_zero",
+            cve_description="cve_description",
+            embargoed=False,
+            statement="statement",
+            cve_id="CVE-2001-0489",
+        )
+        FlawFactory(
+            title="other flaw",
+            comment_zero="comment_zero",
+            cve_description="cve_description",
+            embargoed=True,
+            statement="statement",
+            cve_id="CVE-2001-0494",
+        )
+        FlawFactory(
+            title="flaw with different CVE",
+            comment_zero="comment_zero",
+            cve_description="cve_description",
+            embargoed=False,
+            statement="statement",
+            cve_id="CVE-2008-0514",
+        )
+
+        # Search with djangoql query
+        response = auth_client().get(
+            f'{test_api_uri}/flaws?query=title startswith "flaw"'
+        )
+        assert response.status_code == 200
+        body = response.json()
+        assert body["count"] == 1
+        assert body["results"][0]["cve_id"] == "CVE-2008-0514"
+
+        response = auth_client().get(
+            f'{test_api_uri}/flaws?query=cve_id startswith "CVE-2001" and title = "title"'
```

Comment:
Just curious, are parenthesis nested queries supported? Eg. `(x and y) or z`?

Reply:
Yes, I added a comment to the spike with some advanced searches in case you want to take a look OSIDB-3204